### PR TITLE
[AIRFLOW-3468] Remove KnownEvent(Event)?

### PR DESCRIPTION
**Pull request by Fokko**
The KnownEvent and KnownEventType aren't used by 99% of the companies and therefore we would like to deprecate this for Airflow 2.0. Also, this isn't available in the new RBAC view.

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-3468\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3468
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-3468\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/models/__init__.py [hunk 1]
```diff
@@ -4376,37 +4376,6 @@ def __repr__(self):
         return self.label
 
 
-class KnownEventType(Base):
```
Comment (XD-DENG) [inline airflow/models/__init__.py:4396]: Is an `alembic` file needed? Given tables `known_event_type` and `known_event` are removed.

**Review by feng-tao — CHANGES_REQUESTED** [review]
it needs to have an alembic migration script. And could you send an email to mailing alias to get some feedback about this change in case someone still needs it?

File: airflow/models/__init__.py (hunk 1 above)
Comment (Fokko) [reply airflow/models/__init__.py:4396]: Leaving out an alembic script was a conscious decision. The table will be kept for existing airflow setups, but won't be created for the new ones.

Comment (XD-DENG) [reply airflow/models/__init__.py:4396]: Makes sense to me.

**Comment by Fokko** [comment]
Rebased. @feng-tao I've looked into the Alembic script, but it becomes quite nasty in my opinion. The upgrade will be a `DROP TABLE IF EXISTS`, and the downgrade will recreate the tables which aren't used in Airflow 2.0 anymore.

**Review by feng-tao — APPROVED** [review]

**Comment by feng-tao** [comment]
@Fokko , I see your point. +1

**Comment by feng-tao** [comment]
but CI is failing.

**Comment by feng-tao** [comment]
@Fokko , regarding not having a migration script, I wonder what will happen for the following case:
assume we have three tables: ``A``, ``B``, ``C``, we modify  table ``A`` and provide an alemic migration script(version 1), then delete table ``B`` without a script, then modify table ``C`` with another alemic script(version 2), in this case will the migration(upgrade / downgrade) run successfully from version 1 to  2 or vice versa?  If yes, I am +1 :)

**Comment by Fokko** [comment]
@feng-tao There are no migrations with KnownEvent(Type):
```
MacBook-Pro-van-Fokko:incubator-airflow fokkodriesprong$ grep -R -i knownevent airflow/migrations/versions/
∅
```

I've rebased to fix the flaky mysql test

**Comment by jgao54** [comment]
+1 on deleting under-utilized features :P 

I see that CI/CD is still failing on ```sqlalchemy.exc.OperationalError: (_mysql_exceptions.OperationalError) (3730, "Cannot drop table 'users' referenced by a foreign key constraint 'known_event_ibfk_2' on table 'known_event'.") [SQL: '\nDROP TABLE users']```

I think this is happening when attempting `airflow reset` in the ci/cd scripts, which in turn looks for all the tables that the sqlalchemy engine can find, and then delete them. But since the code for KnownEvent/KnownEventType models is removed, those tables didn't get detected/deleted, so now the `user` table can't get deleted because `known_event` has a reference to it. I think we will need to modify the ci/cd script and explicitly add a `drop table known_event; drop table known_event_type;` now... maybe there's a better way to do that. 

In addition, @Fokko if you do a grep on `known_events` instead of `knownevent`, there are quite a few changes in the alembic scripts, but I don't see any problems with it since they are just creating/upgrading tables.

**Comment by jgao54** [comment]
@feng-tao 
my 2c on alembic scripts example you provided. It should be fine because they are are different tables. Afaik alembic scripts simply carry out sql commands so as long as the underlying upgrade/downgrade are valid, it should run just fine.

**Comment by Fokko** [comment]
Thanks @feng-tao for the clarification. I've removed the `known_event` and `known_event_type` from the migrations and the CI is happy now! 👍

**Review by feng-tao — APPROVED** [review]
@Fokko , good job, and thanks @jgao54 for the info.

**Comment by jgao54** [comment]
Minor concern: I tried to run `airflow resetdb` locally which had a `known_type` table, and still hitting the foreign key constraint error.  Manually deleting those tables beforehand fixes it, I think we should probably document this as well.

Also, would be good to add this change to updating.md (in separate pr) :)

**Comment by bolkedebruin** [comment]
@Fokko would you mind doing a small follow up Pr?

**Pull request merged**